source/pervasive/string.rs

```diff
@@ -247,6 +247,12 @@ impl String {
     {
         self.inner
     }
+
+    #[verifier(external)]
+    pub fn into_rust_string_ref(&self) -> &std::string::String
```

Comment:
Just a suggestion wrt naming:
```suggestion
    pub fn as_rust_string_ref(&self) -> &std::string::String
```
For a borrowed->borrowed conversion (that is not expensive, and exposes a view into the underlying data), the API convention in Rust is to use an `as_*` name https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

I don't know if the `rust_string_ref` part is ideal or not (I _personally_ think it is fine), but I do think maintaining the `as_` portion would likely be better than `into_`